_posts/2024-12-xx-hybrid-search-optimization.md

```diff
@@ -0,0 +1,342 @@
+---
+layout: post
+title: "Optimizing Hybrid Search in OpenSearch"
+authors:
+   - dwrigley
+date: 2024-12-xx
+categories: 
+    - technical-posts
+    - community
+meta_keywords: hybrid query, hybrid search, neural query, lexical search, search relevancy, search result quality optimization 
+meta_description: Tackle the optimization of hybrid search in a systematic way and train models that dynamically predict the best way to run hybrid search in your search application.
+---
+
+# Introduction
+
+[Hybrid search combines lexical and neural search to improve search relevance](https://opensearch.org/docs/latest/search-plugins/hybrid-search); this combination shows promising results across industries and [in benchmarks](https://opensearch.org/blog/semantic-science-benchmarks/).
+
+In OpenSearch 2.18, [hybrid search](https://opensearch.org/docs/latest/search-plugins/hybrid-search/) is a linear combination of the lexical (match query) and neural (kNN) search scores. It first normalizes the scores and then combines them with one of three techniques (arithmetic, harmonic or geometric mean), each of which includes weighting parameters.
+
+The search pipeline configuration is how OpenSearch users define score normalization, combination, and weighting.
+
+# Finding the right hybrid search configuration is hard
+
+The question for a user of hybrid search in OpenSearch is how to choose the normalization and combination techniques and the weighting parameters for their application.
+
+What is best depends strongly on the corpus, on user behavior, and on the application domain – there is no one-size-fits-all solution. 
+
+However, there is a systematic way to arrive at this ideal set of parameters. We call identifying the best set of parameters *global hybrid search optimization*: we identify the best parameter set for all incoming queries; it is “global” because it doesn’t depend on per-query factors. We will cover this approach first before moving on to a dynamic approach that identifies hybrid query parameters individually per query.
+
+# Global hybrid search optimizer
+
+We treat hybrid search configuration as a parameter optimization challenge. The parameters and combinations are:
+
+* Two [normalization techniques: `l2` and `min_max`](https://opensearch.org/blog/How-does-the-rank-normalization-work-in-hybrid-search/)  
+* Three combination techniques: arithmetic mean, harmonic mean, geometric mean  
+* The lexical and neural search weights, values in the range from 0 to 1.
+
+
+With this knowledge we can define a collection of parameter combinations to try out and compare to each other. To follow this path we need three things:
+
+1. Query set: a collection of queries.  
+2. Judgments: a collection of ratings that tell how relevant a result for a given query is.  
+3. Search Metrics: a numeric expression of how well the search system does in returning relevant documents for queries
```

Comment:
Change "Search Metrics: a numeric expression" to "Search Quality Metrics: a numeric expression"